src/modules/position_estimator_inav/position_estimator_inav_main.c

```diff
@@ -571,12 +586,48 @@ int position_estimator_inav_thread_main(int argc, char *argv[])
+					//this flag is not working -->
+					flow_accurate = true; //already checked if flow_q > 0.3
+
+
+					/*calculate offset of flow-gyro using already calibrated gyro from autopilot*/
+					flow_gyrospeed[0] = flow.gyro_x_rate_integral/(float)flow.integration_timespan*1000000.0f;
+					flow_gyrospeed[1] = flow.gyro_y_rate_integral/(float)flow.integration_timespan*1000000.0f;
+					flow_gyrospeed[2] = flow.gyro_z_rate_integral/(float)flow.integration_timespan*1000000.0f;
+					
+					//moving average
+					if (n_flow >= 100) {
+						gyro_offset_filtered[0] = flow_gyrospeed_filtered[0] - att_gyrospeed_filtered[0];
+						gyro_offset_filtered[1] = flow_gyrospeed_filtered[1] - att_gyrospeed_filtered[1];
+						gyro_offset_filtered[2] = flow_gyrospeed_filtered[2] - att_gyrospeed_filtered[2];
+						n_flow = 0;
+						flow_gyrospeed_filtered[0] = 0.0f;
+						flow_gyrospeed_filtered[1] = 0.0f;
+						flow_gyrospeed_filtered[2] = 0.0f;
+						att_gyrospeed_filtered[0] = 0.0f;
+						att_gyrospeed_filtered[1] = 0.0f;
+						att_gyrospeed_filtered[2] = 0.0f;
+					} else {
+						flow_gyrospeed_filtered[0] = (flow_gyrospeed[0] + n_flow * flow_gyrospeed_filtered[0]) / (n_flow + 1);
+						flow_gyrospeed_filtered[1] = (flow_gyrospeed[1] + n_flow * flow_gyrospeed_filtered[1]) / (n_flow + 1);
+						flow_gyrospeed_filtered[2] = (flow_gyrospeed[2] + n_flow * flow_gyrospeed_filtered[2]) / (n_flow + 1);
+						att_gyrospeed_filtered[0] = (att.pitchspeed + n_flow * att_gyrospeed_filtered[0]) / (n_flow + 1);
+						att_gyrospeed_filtered[1] = (att.rollspeed + n_flow * att_gyrospeed_filtered[1]) / (n_flow + 1);
+						att_gyrospeed_filtered[2] = (att.yawspeed + n_flow * att_gyrospeed_filtered[2]) / (n_flow + 1);
+						n_flow++;
+					}
+					
+
+					/*yaw compensation (flow sensor is not in center of rotation) -> params in QGC*/
+					yaw_comp[0] = params.flow_module_offset_x * (flow_gyrospeed[2] - gyro_offset_filtered[2]);
+					yaw_comp[1] =  - params.flow_module_offset_y * (flow_gyrospeed[2] - gyro_offset_filtered[2]);
+
 
 					/* convert raw flow to angular flow (rad/s) */
 					float flow_ang[2];
-					//todo check direction of x und y axis
-					flow_ang[0] = flow.pixel_flow_x_integral/(float)flow.integration_timespan*1000000.0f;//flow.flow_raw_x * params.flow_k / 1000.0f / flow_dt;
-					flow_ang[1] = flow.pixel_flow_y_integral/(float)flow.integration_timespan*1000000.0f;//flow.flow_raw_y * params.flow_k / 1000.0f / flow_dt;
+					//calculate flow [rad/s] and compensate for rotations (and offset of flow-gyro)
+					flow_ang[0] = (flow.pixel_flow_x_integral - flow.gyro_x_rate_integral)/(float)flow.integration_timespan*1000000.0f + gyro_offset_filtered[0] - yaw_comp[0];//flow.flow_raw_x * params.flow_k / 1000.0f / flow_dt;
+					flow_ang[1] = (flow.pixel_flow_y_integral - flow.gyro_y_rate_integral)/(float)flow.integration_timespan*1000000.0f + gyro_offset_filtered[1] - yaw_comp[1];//flow.flow_raw_y * params.flow_k / 1000.0f / flow_dt;
```

Comment:
Because we're subtracting flow.gyro_..._rate_integral which has an offset and gyro_offset_filtered[...] has already been divided by the integration_timespan. Hope this answers your question.